### PR TITLE
[BE] Task #208203 Edit-requests create and update changes

### PR DESCRIPTION
**Pull request by GitVivekHub**
# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

### CODE REVIEW

File: src/src/modules/edit-requests/edit-requests.service.ts [hunk 1]
```diff
@@ -25,20 +25,31 @@ export class EditRequestService {
...
+			'edit_req_for_context',
+			'edit_req_by',
+			'fields',
+		];
+
+		// Check required fields
+		const missingRequiredField = requiredFields.find(
+			(field) =>
+				!body[field] ||
+				(field === 'fields' && body[field].length === 0) ||
```
Comment (sagarkoshti1990) [inline src/src/modules/edit-requests/edit-requests.service.ts:41]: add is array

File: src/src/modules/edit-requests/edit-requests.service.ts [hunk 2]
```diff
@@ -25,20 +25,31 @@ export class EditRequestService {
...
+				body[field] === '',
+		);
+
+		if (missingRequiredField) {
+			return res.json({
+				status: 400,
```
Comment (sagarkoshti1990) [inline src/src/modules/edit-requests/edit-requests.service.ts:47]: status code `422`

File: src/src/modules/edit-requests/edit-requests.service.ts [hunk 3]
```diff
@@ -127,12 +145,24 @@ export class EditRequestService {
...
+		if (body?.status == 'approved' && body?.fields?.length == 0) {
+			return res.json({
+				status: 400,
+				message: 'INVALID_PARAMETERS',
+				data: {},
+			});
+		}
+
+		if (body?.status == 'closed') {
+			body.fields = '[]';
```
Comment (sagarkoshti1990) [inline src/src/modules/edit-requests/edit-requests.service.ts:157]: `body.fields = '[]';` change to `body.fields = [];`

**Review by sagarkoshti1990 — CHANGES_REQUESTED** [review]

File: src/src/modules/edit-requests/edit-requests.service.ts (hunk 1 above)
Comment (GitVivekHub) [reply src/src/modules/edit-requests/edit-requests.service.ts:41]: resolved

File: src/src/modules/edit-requests/edit-requests.service.ts (hunk 3 above)
Comment (GitVivekHub) [reply src/src/modules/edit-requests/edit-requests.service.ts:157]: resolved

File: src/src/modules/edit-requests/edit-requests.service.ts (hunk 2 above)
Comment (GitVivekHub) [reply src/src/modules/edit-requests/edit-requests.service.ts:47]: resolved

**Review by sagarkoshti1990 — APPROVED** [review]

**Pull request merged**